core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1324,34 +1283,39 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
       workspaceId <- loadV2WorkspaceId(workspaceName)
       results <- Future.traverse(input) {
         case WorkspaceCatalog(email, true) =>
-          toFutureTry(
-            samDAO.addUserToPolicy(SamResourceTypeNames.workspace,
-                                   workspaceId,
-                                   SamWorkspacePolicyNames.canCatalog,
-                                   email,
-                                   ctx
+          samDAO
```

Comment:
You can chain as many `.map` calls onto a future as you like, and it won't actually throw the exception until you try to extract the value.
So as long as the `recover` is before the `<-` it's fine.
I think you might even be able to do it at the `yield`, since the `for` comprehension is just syntactic sugar for `.map`.

The actual motivation for this was just to get rid of the `toFutureTry` to eliminate an extra trait that was needlessly complicating things.